### DISCUSSION TITLE
Prevent accidental activation of Additional Features

### DISCUSSION THREAD

**Discussion by Lenbok** [manna-harbour/miryoku#93] [Q&A] [answered]
I am just learning miryoku and so I am often mistakenly using finger combos from my previous layout. One thing that has got me a few times is accidentally activating functions on the "Additional features" layer. These put the keyboard into non-obvious states (and in some cases can only be undone via keyboard reset), so I think it might be best if this layer were not enabled fresh out of the box (some users may never need the functionality in this layer anyway).

**Comment by manna-harbour** [comment]
https://github.com/manna-harbour/miryoku/tree/master/docs/reference#additional-features

Those keys should not be hit in normal use, but it is possible to hit them accidentally, so something needs to be added to prevent that.

I've been considering a few options:

- Build time option:
  - Disable by default, enable with option.
  - Enable by default, disable with option.
- Always enabled but no action if tapped:
  - Require hold.
  - Require to be combined with a mod.
  - Require multiple taps.

Combining with shift is consistent with [shift functions on Media](https://github.com/manna-harbour/miryoku/tree/master/docs/reference#media) and with [caps word / caps lock](https://github.com/manna-harbour/miryoku_zmk#caps-word), but would be awkward, and impossible for functions on the same finger as shift (Base and Cur).

I favour always enabled but no action if tapped, and either require a hold (say 0.5s) or multiple taps (say double tap). If anyone has Opinions please share!

**Comment by manna-harbour** [comment, marked as answer]
Additional Features now require double tap to activate on Miryoku ZMK ([except for Boot](https://github.com/manna-harbour/miryoku_zmk#double-tap-boot)). This will be added to Miryoku QMK and Miryoku KMonad.

**Reply by manna-harbour** [reply]
This in now available for Miryoku QMK in a temporary branch and will be added to the main branch soon: https://github.com/manna-harbour/miryoku_qmk/tree/td-tmp.

**Reply by manna-harbour** [reply]
This in now available for Miryoku QMK and Miryoku KMonad.
